Mathlib/LinearAlgebra/Matrix/Zpow.lean

```diff
@@ -285,19 +287,23 @@ theorem zpow_mul' (A : M) (h : IsUnit A.det) (m n : ℤ) : A ^ (m * n) = (A ^ n)
   rw [mul_comm, zpow_mul _ h]
 #align matrix.zpow_mul' Matrix.zpow_mul'
 
+/-
+porting note: TODO: is this still needed, or should just the norm_cast be removed?
 @[simp, norm_cast]
 theorem coe_units_zpow (u : Mˣ) : ∀ n : ℤ, ((u ^ n : Mˣ) : M) = u ^ n
-  | (n : ℕ) => by rw [_root_.zpow_coe_nat, zpow_ofNat, Units.val_pow_eq_pow_val]
+  | (n : ℕ) => by rw [_root_.zpow_coe_nat]
   | -[k+1] => by
     rw [zpow_negSucc, zpow_negSucc, ← inv_pow, u⁻¹.val_pow_eq_pow_val, ← inv_pow', coe_units_inv]
 #align matrix.coe_units_zpow Matrix.coe_units_zpow
+-/
+#noalign matrix.coe_units_zpow
 
 theorem zpow_ne_zero_of_isUnit_det [Nonempty n'] [Nontrivial R] {A : M} (ha : IsUnit A.det)
     (z : ℤ) : A ^ z ≠ 0 := by
-  have := ha.det_zpow z
-  contrapose! this
-  rw [this, det_zero ‹_›]
-  exact not_isUnit_zero
+  have := IsUnit.det_zpow ha z
```

Comment:
This change can be reverted, it's due to the `#align` above not being fixed properly

Reply:
yes, except that 
`contrapose! this`  still does not work